modules/ob-authentication-to-container-registries.adoc

```diff
@@ -0,0 +1,47 @@
+// This module is included in the following assembly:
+//
+// builds/authenticating-build-images.adoc
+
+:_content-type: PROCEDURE
+[id="ob-authentication-to-container-registries_{context}"]
+= Authentication to container registries
+
+To push images to a private container registry, you must define a secret in the `OpenShift-Builds` namespace and then reference it in your `Build` custom resource (CR).
+
+
+.Procedure
+. Run the following command to generate a secret:
++
+[source,terminal]
+----
+oc --namespace <namespace> create secret docker-registry <container_registry_secret_name> \
+  --docker-server=<registry_host> \ <1>
+  --docker-username=<username> \ <2>
+  --docker-password=<password> \ <3>
+  --docker-email=<email_address>
+---- 
+<1> The `<REGISTRY_HOST>` value must be set to `\https://index.docker.io/v1/`
+<2> The `<USERNAME>` is the Docker ID
+<3> The `<PASSWORD>` value can be your docker hub password or an access token
+
+. Run the following command to annotate the secret:
++
+[source,terminal]
+----
+oc --namespace <namespace> annotate secrets <container_registry_secret_name> build.shipwright.io/referenced.secret='true'
+---- 
+
+. Set the value of the `spec.output.pushSecret` field to the secret name in your `Build` CR:
++
+[source,yaml]
+----
+apiVersion: shipwright.io/v1beta1
+kind: Build
+metadata:
+  name: buildah-golang-build
+  # ...
+  output:
+    image: docker.io/foobar/sample:latest
```

Comment:
Per IBM SG, do not use `foo`, `foobar`, etc in examples.

Reply:
@apoorvajagtap Any alternative to replace foo and foobar from examples?

We need to replace one more instance of `foo` here: https://63890--docspreview.netlify.app/openshift-builds/latest/builds/configuring-build-runs#ob-defining-the-build-specification_configuring-build-runs

Reply:
I don't think we have any image named `sample` in `docker.io`. We can have something generic for the whole image (this way we'll also eliminate the reference to docker.io specifically). Maybe something like <your_image> whichever format you find best.